src/test/java/core/basesyntax/service/impl/FileDataParserCsvImplTest.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.FileDataParserCsvImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FileDataParserCsvImplTest {
+    private FileDataParser fileDataParser;
+
+    @BeforeEach
+    public void setUp() {
+        fileDataParser = new FileDataParserCsvImpl();
+    }
+
+    @Test
+    public void parseData_validData_ok() {
+        List<String> dataFromFile = Arrays.asList("b,apple,10", "s,banana,20",
+                "p,pear,30", "r,melon,40");
+        List<FruitTransaction> expected = Arrays.asList(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 10),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 20),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "pear", 30),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "melon", 40)
+        );
+        List<FruitTransaction> actual = fileDataParser.parseData(dataFromFile);
+        assertEquals(expected, actual, "Parsed transactions do not match the expected output.");
+    }
+
+    @Test
+    public void parseData_nullData_notOk() {
+        assertThrows(NullPointerException.class, () -> fileDataParser.parseData(null),
+                "NullPointerException should be thrown if the input list is null.");
```

Comment:
NPE is not good practice, better handle this exception for not throwing

Reply:
refactored